Feature/tls for mysql

daniellowtw opened the pull request.

Commit:
Add TLS for mysql

Commit:
Remove spurious println

daniellowtw commented:
@mwitkow PTAL

daniellowtw commented:
@jonny-improbable

mwitkow commented on conf/defaults.ini:
```diff
@@ -62,6 +62,13 @@ password =
 ssl_mode = disable
 
+# For "mysql" only, either true, false, skip-verify
+tls_mode = false
```
can you reuse the `ssl_mode` for this?

daniellowtw replied:
The dsn structure for having ssl is different for postgres and mysql. They accept different values. I could make it look similar by doing mysql_ssl_mode

mwitkow commented on conf/defaults.ini:
```diff
@@ -62,6 +62,13 @@ password =
+ca_path =/etc/mysql/ca.pem
+client_key_path =/etc/mysql/client-key.pem
+client_cert_path =/etc/mysql/client-cert.pem
+server_name =localhost
```
name this something like `server_cert_name`

mwitkow commented:
Please update docs/sources/installation/configuration.md to document the changes you made

mwitkow commented on pkg/services/sqlstore/sqlstore.go:
```diff
@@ -115,6 +126,15 @@ func getEngine() (*xorm.Engine, error) {
 		cnnstr = fmt.Sprintf("%s:%s@tcp(%s)/%s?charset=utf8",
 			DbCfg.User, DbCfg.Pwd, DbCfg.Host, DbCfg.Name)
+		if mysqlConfig.UseTls != "false" {
+			fmt.Printf("%+v", mysqlConfig)
```
remove the printf

mwitkow commented on pkg/services/sqlstore/tls_mysql.go:
```diff
@@ -0,0 +1,36 @@
+package sqlstore
+import (
+	"crypto/tls"
```
you should be using tabs for go, right?

daniellowtw replied:
It is using tabs

mwitkow commented on pkg/services/sqlstore/tls_mysql.go:
```diff
@@ -0,0 +1,36 @@
+	rootCertPool := x509.NewCertPool()
+	pem, err := ioutil.ReadFile(config.CaPath)
+	if err != nil {
+		return nil, err
```
improve error reporting throughout this code:

`fmt.Errorf("could not read DB CA Cert path: %v", err)`

mwitkow commented on pkg/services/sqlstore/tls_mysql.go:
```diff
@@ -0,0 +1,36 @@
+		tlsConfig.ServerName = config.ServerName
+	}
+
+	if config.UseTls == "skip-verify" {
```
make this map to an existing `ssl_mode`

mwitkow commented:
ROFL

Commit:
Use ssl_mode for mysql and add docs

reverted default configs to use sqlite
remove spurious printf
add docs for the new parameters in config
rename parameters in the config for readability

Commit:
Tolerate ssl_mode without client authentication

Client cert is not necessary for a SSL connection. So we tolerate
failure if client cert is not provided.
Improve error message if missing server_cert_name and mode is not
skip-verify.

daniellowtw commented:
PTAL
